github/repos_deployments_test.go

```diff
@@ -89,6 +89,28 @@ func TestRepositoriesService_CreateDeployment(t *testing.T) {
 	}
 }
 
+func TestRepositoriesService_DeleteDeployment(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/deployments/1", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "DELETE")
+	})
+
+	_, err := client.Repositories.DeleteDeployment(context.Background(), "o", "r", 1)
+	if err != nil {
+		t.Errorf("Repositories.DeleteDeployment returned error: %v", err)
+	}
+
+	resp, err := client.Repositories.DeleteDeployment(context.Background(), "o", "r", 2)
+	if err == nil {
+		t.Error("Repositories.DeleteDeployment should return an error")
+	}
+	if resp.StatusCode != http.StatusNotFound {
+		t.Error("Repositories.DeleteDeployment should return a 404 status")
```

Comment:
From the API docs, it looks like it should return `204 No Content` on a successful request?

Reply:
Thank you, @nightlark. 
@nomonamo - once you add the line above, then this can be changed to `http.StatusNoContent` and the 404 can be changed to 204.

Reply:
In this case I was just adding a test for a non-existent deployment (id: 2), in an attempt to increase test coverage.